### DISCUSSION TITLE
How to configure Github OAuth with Supbase?

### DISCUSSION THREAD

**Discussion by raqibhayder** [supabase/supabase#2014] [Questions]
I am trying to build a chrome extension with Github OAuth using Supbase. The extension is a sidebar on Github.

From our chrome extension I start the authentication flow like so but the user is always null.

``` javascript
async function auth() {
  const { user, error } = await supabase.auth.signIn({
    provider: "github"
  });
  if (error) console.error(error);
  else console.log(user);
}
```

I think this might have something to do with the redirect urls and callbacks.  I have the following set up on Github and Supabase w.r.t to redirect URL.

Supabase:
- Site URL: www.github.com
- Additional Redirect URLs: 

Github:
- Homepage: https://www.github.com
- Authorization callback URL: https://[our-app-name].supabase.co/auth/v1/callback

🙂

**Comment by inian** [comment]
Hi @raqib-hayder, @burggraf published this Youtube video recently on setting up Github Authentication - https://www.youtube.com/watch?v=RlfCIvmimSA

Can you verify if you have set up your Github application properly?

**Reply by burggraf** [reply]
Site URL in Supabase should be your site, not github.com.

**Reply by raqibhayder** [reply]
We are a chrome extension (a sidebar) that sits on top of Github. Is there a specific way to configure the URL for chrome extensions?

**Reply by burggraf** [reply]
Oooh, I'm not sure about that.  Where should Supabase auth send the result?

**Reply by silentworks** [reply]
@raqib-hayder I looked at Octotree to see how they are handling this seen as they are a chrome extension and uses Github to log in. They have a separate page on a hosted website which is used as the Site URL to handle the redirect.

**Reply by raqibhayder** [reply]
@burggraf: That is what I am not exactly sure about. 

@silentworks: I was looking at [this](https://www.youtube.com/watch?v=0RpOzhjhTzA) tutorial for firebase on youtube. Basically, that is the auth flow that I want to replicate. They talk about the callbacks [here](https://youtu.be/0RpOzhjhTzA?t=1101). 

I really appreciate the eager help from everyone. 🙏🏽

**Reply by burggraf** [reply]
I think that video complicates things because they're using another third party library as a layer on top of Firebase auth.  It should be simpler if you just create a site to do basic auth with Supabase then use that in a window like he did in the video.

**Comment by burggraf** [comment]
Ah, nice.  Makes sense.  Thanks so much for following up here.

**Comment by raqibhayder** [comment]
Perfect. Will do that and post an update. Thanks a lot for all your help.
